api/services/SiteMembershipCreator.js

```diff
@@ -52,8 +52,15 @@ const createSiteMembership = ({ user, siteParams }) => {
   ).then(() => site);
 };
 
-const revokeSiteMembership = ({ user, site, userId }) =>
-  GitHub.checkPermissions(user, site.owner, site.repository)
+const revokeSiteMembership = ({ user, site, userId }) => {
+  if (userId === user.id && user.username.toLowerCase() === site.owner.toLowerCase()) {
```

Comment:
I think to prevent a user from removing the site owner you'd first need grab the user model associated with `userId` and then check that user's username field against the `site.owner`.

Reply:
Yep, I was planning on adding that in one of the follow up PRs since that criteria wasn't really established until yesterday afternoon, I'll add it now!